src/iso15118/d20/state/authorization.cpp

```diff
@@ -93,14 +98,29 @@ Result Authorization::feed(Event ev) {
         return {};
     }
 
+    if (ev == Event::TIMEOUT) {
+        const auto timeout = m_ctx.get_active_timeout();
+        if (timeout and *timeout == d20::TimeoutType::ONGOING) {
+            timeout_ongoing_reached = true;
+        }
+        return {};
+    }
+
     if (ev != Event::V2GTP_MESSAGE) {
         return {};
     }
 
     const auto variant = m_ctx.pull_request();
 
     if (const auto req = variant->get_if<message_20::AuthorizationRequest>()) {
-        const auto res = handle_request(*req, m_ctx.session, authorization_status);
+
+        if (first_req_msg) {
+            // TODO(SL): For ExternalPayment the timeout should at least 3 mins?
+            m_ctx.start_timeout(d20::TimeoutType::ONGOING, 55000); // TODO(sl): Adding const
```

Comment:
Not sure if these TODOs were forgotton or aren't meant for this PR

Reply:
Forgotten and worked on them